### DISCUSSION TITLE
Full list of atlas.hcl properties

### DISCUSSION THREAD

**Discussion by malliaridis** [ariga/atlas#1203] [Q&A]
Hey there, I was looking for a full list of the configuration properties that can be provided in the `atlas.hcl` file, but I couldn't find any page in the documentation. The closest I get to the options available is the page [Atlas Project Files](https://atlasgo.io/atlas-schema/projects#project-files).

So my question: Is there somewhere a complete list of properties I can configure in `atlas.hcl`? Or is there a concept behind the property definition that keeps it flexible? The question applies to nested properties too (like properties configurable inside an `env` or `lint` block).

**Comment by a8m** [comment]
Hey @malliaridis! These are great questions.

We plan to release a new version with plenty of new features that were added to our `atlas.hcl` configuration/project file, and it will be documented on the website by me soon. I'll update this discussion after it's done. 

Regarding the second question, @rotemtam and I just discussed this subject yesterday, and in the upcoming weeks, we're going to work on moving the Atlas schema to be type-safe with proper editor support (language server) with generated documentation. This will also solve the first question.
